reproducibility_project/src/engines/gomc/project.py

```diff
@@ -1,23 +1,2174 @@
+    print(
+        "# Running test if the pymbar_stabilized_equilb_design_ensemble is stable."
+    )
+    print("#**********************")
+
+    return job.doc.stable_equilb_design_ensemble
+
+
+# ******************************************************
+# ******************************************************
+# Use pymbar to evaluate if the system came to equilibrium based on
+# the density (NPT or NVT -> box 0 density, GEMC or GCMC box 1 density) (end)
+# ******************************************************
+# ******************************************************
+
+# ******************************************************
+# ******************************************************
+# equilb NPT or GEMC-NVT - starting the GOMC simulation (start)
+# ******************************************************
+# ******************************************************
+@Project.pre(lambda j: j.sp.engine == "gomc")
+@Project.pre(part_4b_job_equilb_NVT_completed_properly)
+@Project.pre(part_2c_equilb_design_ensemble_control_file_written)
+@Project.post(part_3c_output_equilb_design_ensemble_started)
+@Project.post(part_4c_job_equilb_design_ensemble_completed_properly)
+@Project.post(pymbar_stabilized_equilb_design_ensemble)
+@Project.operation.with_directives(
+    {
+        "np": lambda job: ff_info_dict.get(job.sp.forcefield_name).get("ncpu"),
+        "ngpu": lambda job: ff_info_dict.get(job.sp.forcefield_name).get(
+            "ngpu", 0
+        ),
+        "memory": memory_needed,
+        "walltime": walltime_gomc_hr,
+    }
+)
+@flow.with_job
+def run_equilb_ensemble_gomc_command(job):
+    """Run the gomc equilb_ensemble simulation."""
+    for run_equilb_ensemble_i in range(
+        job.doc.equilb_design_ensemble_number, equilb_design_ensemble_max_number
+    ):
+        print("#**********************")
+        print("# Started the run_equilb_ensemble_gomc_command function.")
+        print("#**********************")
+
+        if (
+            job.doc.equilb_design_ensemble_number
+            >= equilb_design_ensemble_max_number
+        ):
+            job.doc.equilb_design_ensemble_max_number_under_limit = False
+
+        elif (
+            job.doc.stable_equilb_design_ensemble is False
+            and job.doc.equilb_design_ensemble_max_number_under_limit is True
+        ):
+
+            control_file_name_str = job.doc.equilb_design_ensemble_dict[
+                str(job.doc.equilb_design_ensemble_number)
+            ]["output_name_control_file_name"]
+
+            print(f"Running simulation job id {job}")
+            run_command = "{}/{} +p{} {}.conf > out_{}.dat" "".format(
+                str(job.doc.gomc_binary_path),
+                str(job.doc.equilb_design_ensemble_gomc_binary_file),
+                str(ff_info_dict.get(job.sp.forcefield_name).get("ncpu")),
+                str(control_file_name_str),
+                str(control_file_name_str),
+            )
+
+            exec_run_command = subprocess.Popen(
```

Comment:
This method may be possible in the future.  However, I have already tested the current implementation and do not want to modify and test everything again since the end product would be the same anyway, and we are not 100% sure it will work if changed.